pyvlx/api/command_send.py

```diff
@@ -66,13 +68,21 @@ async def handle_frame(self, frame: FrameBase) -> bool:
             return True
         return False
 
+    async def send(self) -> None:
+        """Send frame to KLF200."""
+        await self.do_api_call()
+        if not self.success:
+            raise PyVLXException("Unable to send command")
+
     def request_frame(self) -> FrameCommandSendRequest:
         """Construct initiating frame."""
-        self.session_id = get_new_session_id()
-        return FrameCommandSendRequest(
-            node_ids=[self.node_id],
-            parameter=self.parameter,
-            active_parameter=self.active_parameter,
-            session_id=self.session_id,
-            **self.functional_parameter
-        )
+        if self.frame is None:
+            self.session_id = get_new_session_id()
+            self.frame = FrameCommandSendRequest(
```

Comment:
What is the reason for storing the frame within the class? 

If you change any members of the class, i would assume `request_frame` (or probably better `construct_frame` ) would create&return a freshly constructed frame. Introducing caching here is hidden logic and might cause hard-to-debug bug classes.

Reply:
Oh, that is a residual from a trail to fix the "Unable to send command" issue before implementing a Semaphore. First I tried to simply resent the same and in order to avoid a new session_id, I put them into a class: 
https://github.com/pawlizio/pyvlx/commit/1a652de54d6487f0a79dd65eab011cb27bf6cee3

I will revert this.